src/goto-instrument/contracts/dynamic-frames/dfcc_is_cprover_symbol.cpp

```diff
@@ -14,9 +14,142 @@ Date: March 2023
 #include <util/prefix.h>
 #include <util/suffix.h>
 
-bool dfcc_is_cprover_symbol(const irep_idt &id)
+#include <unordered_set>
+
+static std::unordered_set<irep_idt> function_symbols;
+static std::unordered_set<irep_idt> static_symbols;
```

Comment:
Both of these could safely be moved into their respective `dfcc_is_...` functions: the `init` functions would just have to take a reference to the set as a parameter.